src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -910,28 +910,80 @@ namespace cryptonote
     crypto::key_derivation derivation;
-    bool r = acc.get_device().generate_key_derivation(tx_pub_key, acc.m_view_secret_key, derivation);
-    CHECK_AND_ASSERT_MES(r, false, "Failed to generate key derivation");
+    if (!generate_key_derivation(acc, tx_pub_key, derivation))
+    {
+      return false;
+    }
     crypto::public_key pk;
-    r = acc.get_device().derive_public_key(derivation, output_index, acc.m_account_address.m_spend_public_key, pk);
-    CHECK_AND_ASSERT_MES(r, false, "Failed to derive public key");
-    if (pk == out_key.key)
-      return true;
-    // try additional tx pubkeys if available
+    if (!derive_public_key(acc, derivation, output_index, pk))
+    {
+      return false;
+    }
+    return is_pk_out_key(derivation, pk, out_key);
+  }
+
+  bool is_out_to_acc_additional(const account_keys& acc, const txout_to_key& out_key, const std::vector<crypto::public_key>& additional_tx_pub_keys, size_t output_index)
+  {
     if (!additional_tx_pub_keys.empty())
     {
       CHECK_AND_ASSERT_MES(output_index < additional_tx_pub_keys.size(), false, "wrong number of additional tx pubkeys");
-      r = acc.get_device().generate_key_derivation(additional_tx_pub_keys[output_index], acc.m_view_secret_key, derivation);
-      CHECK_AND_ASSERT_MES(r, false, "Failed to generate key derivation");
-      r = acc.get_device().derive_public_key(derivation, output_index, acc.m_account_address.m_spend_public_key, pk);
-      CHECK_AND_ASSERT_MES(r, false, "Failed to derive public key");
-      return pk == out_key.key;
+      const crypto::public_key& tx_pub_key_additional = additional_tx_pub_keys[output_index];
+      return is_pk_out_key(acc, out_key, tx_pub_key_additional, output_index);
     }
     return false;
   }
+
+  bool is_out_to_acc(const account_keys& acc, const txout_to_key& out_key, const crypto::key_derivation & derivation, const std::vector<crypto::public_key>& additional_tx_pub_keys, size_t output_index)
+  {
+    if (is_pk_out_key_deriv(acc, out_key, derivation, output_index))
+    {
+       return true;
+    }
+    // try additional tx pubkeys if available
+    return is_out_to_acc_additional(acc, out_key, additional_tx_pub_keys, output_index);
+  }
+
+  bool is_out_to_acc(const account_keys& acc, const txout_to_key& out_key, const crypto::public_key& tx_pub_key, const std::vector<crypto::public_key>& additional_tx_pub_keys, size_t output_index)
+  {
+    if (is_pk_out_key(acc, out_key, tx_pub_key, output_index))
+    {
+       return true;
+    }
+    // try additional tx pubkeys if available
+    return is_out_to_acc_additional(acc, out_key, additional_tx_pub_keys, output_index);
+  }
+
+  bool generate_key_derivation(const account_keys& acc, const crypto::public_key& tx_pub_key, crypto::key_derivation & derivation)
+  {
+    const bool r = acc.get_device().generate_key_derivation(tx_pub_key, acc.m_view_secret_key, derivation);
+    CHECK_AND_ASSERT_MES(r, false, "Failed to generate key derivation");
+    return r;
```

Comment:
This should always `return true` given the above check.

Reply:
Provided that the macro works fine, but can be changed to true, no prob.